R/lasso-development.R

```diff
@@ -165,181 +165,261 @@
       # creating the model matrix.
-      private$modFmla <- as.formula(paste("~",paste(names(private$dfTrainTemp[ ,!(colnames(private$dfTrainTemp) == self$params$predictedCol)]),
-                                           collapse = "+")))
-
+      private$modFmla <-
+        as.formula(paste("~", paste(
+          names(private$dfTrainTemp[, !(colnames(private$dfTrainTemp) == self$params$predictedCol)]),
+          collapse = "+"
+        )))
+      
+      # print('saving test file...')
+      # modFmla <- private$modFmla
+      # save(modFmla, file = "modFmla_TEST.rda")
+      
       # Create the model matrix, without the intercept column, to be used in the
       # grouped Lasso function.
-      private$modMat <- model.matrix(private$modFmla, data = private$dfTrainTemp)[,-1]
-
+      private$modMat <-
+        model.matrix(private$modFmla, data = private$dfTrainTemp)[, -1]
+      
       # Make sure the dependent variable is numeric in both the train and
       # test set. If it is a factor, the first level alphabetically will be
       # set to 0 and the other level will be set to 1.
       if (is.factor(private$dfTrainTemp[[self$params$predictedCol]])) {
         private$dfTrainTemp[[self$params$predictedCol]] <-
-          ifelse(private$dfTrainTemp[[self$params$predictedCol]] == levels(private$dfTrainTemp[[self$params$predictedCol]])[1],0,1)
+          ifelse(private$dfTrainTemp[[self$params$predictedCol]] == levels(private$dfTrainTemp[[self$params$predictedCol]])[1],
+                 0,
+                 1)
       }
-
+      
       if (is.factor(private$dfTestTemp[[self$params$predictedCol]])) {
         private$dfTestTemp[[self$params$predictedCol]] <-
-          ifelse(private$dfTestTemp[[self$params$predictedCol]] == levels(private$dfTestTemp[[self$params$predictedCol]])[1],0,1)
+          ifelse(private$dfTestTemp[[self$params$predictedCol]] == levels(private$dfTestTemp[[self$params$predictedCol]])[1], 0, 1)
       }
-
+      
       # Creating the groups for the grouped Lasso model.
       # Factor variables have a group that is one less than the number of levels.
       # Everything else has length one.
       # The length of this vector should be the same as the number of columns in modMat.
-      private$group <- rep(1:(ncol(private$dfTrainTemp) - 1 ),
-                  times <- sapply(private$dfTrainTemp[ ,!(colnames(private$dfTrainTemp) == self$params$predictedCol)],
-                               function(x) ifelse(is.factor(x), length(levels(x)) - 1, 1)))
-
+      private$group <-
+        rep(1:(ncol(private$dfTrainTemp) - 1),
+            times <-
+              sapply(private$dfTrainTemp[, !(colnames(private$dfTrainTemp) == self$params$predictedCol)],
+                     function(x)
+                       ifelse(is.factor(x), length(levels(x)) - 1, 1)))
+      
       if (length(private$group) != ncol(private$modMat)) {
         stop('There is a mismatch in group definition and model matrix definition')
```

Comment:
While we're in here, would you mind making this error more helpful? Perhaps give the user something to look for or use a unit test to make sure it doesn't occur?